crates/burn-train/src/metric/base.rs

```diff
@@ -56,6 +59,84 @@ pub trait Metric: Send + Sync {
     fn clear(&mut self);
 }
 
+/// Classification Metric trait
+///
+/// Requires implementation [Metric](Metric)<Input = ClassificationInput<B>>
+pub trait ClassificationMetric<B: Backend>: Metric<Input = ClassificationInput<B>> {
+    /// Sets threshold. Default 0.5
+    fn with_threshold(self, threshold: f64) -> Self;
+    /// Sets average type. Default Micro
+    fn with_average(self, average: ClassificationAverage) -> Self;
+}
```

Comment:
Hmmm I don't think a new trait is necessary for that. I'd remove it from the current implementation.

Reply:
Removed!